application-ckeditor-plugins/src/main/resources/xwiki-image/plugin.js

```diff
@@ -31,8 +31,12 @@
     beforeInit: function(editor) {
       editor.on('widgetDefinition', function (event) {
         var widgetDefinition = event.data;
-        if (widgetDefinition.name === "image" && widgetDefinition.dialog === "image2") {
-          this.overrideImageWidget(editor, widgetDefinition);
+        if (widgetDefinition.name === "image") {
```

Comment:
Only the [Enhanced Image](https://ckeditor.com/cke4/addon/image2) plugin has support for captions AFAIK. The simple [Image](https://ckeditor.com/cke4/addon/image) plugin doesn't have it, so I think you should keep both conditions. You want to configure the allowed caption content only for the Enhanced Image plugin, whose dialog is ``image2``.